command/agent/agent_test.go

```diff
@@ -1085,6 +1085,39 @@ func TestServer_Reload_TLS_DowngradeFromTLS(t *testing.T) {
 	assert.True(agent.config.TLSConfig.IsEmpty())
 }
 
+func TestServer_Reload_VaultConfig(t *testing.T) {
+	ci.Parallel(t)
+
+	logger := testlog.HCLogger(t)
+
+	agentConfig := &Config{
+		TLSConfig: &config.TLSConfig{},
+		Vault: &config.VaultConfig{
+			Enabled:   pointer.Of(true),
+			Token:     "vault-token",
+			Namespace: "vault-namespace",
+		},
+	}
+
+	agent := &Agent{
+		auditor: &noOpAuditor{},
+		logger:  logger,
+		config:  agentConfig,
+	}
+
+	newConfig := &Config{
+		TLSConfig: &config.TLSConfig{},
+		Vault: &config.VaultConfig{
+			Enabled:   pointer.Of(true),
+			Token:     "vault-token",
+			Namespace: "vault-namespace",
+		},
```

Comment:
Is `newConfig` supposed to be different from `agentConfig`? It think you would want to test if `newConfig` was applied after a `Reload`?

Reply:
Oh good call. And I should be checking `agent.GetConfig().Vault` here too to get the updated config.

Reply:
Well as it turns out, reloading the config _doesn't_ set the server config we keep in the server struct. That still holds on to the old version of the config as far as I can tell, because we're never supposed to read from it at runtime. That seems like a bug, tbh, but unrelated and there's a lot of threads to pull on that.

We do overwrite the `VaultConfig` that we write into the server's `VaultClient`, but I can't get to the one in the agent in this test. I've added an accessor method so that we can access it in the `nomad/server_test.go` test at least.

Reply:
Done in [9692586](https://github.com/hashicorp/nomad/pull/14298/commits/9692586e77c3121737f6fe2c14fd6037520410c4)